test/src/test/java/org/corfudb/runtime/concurrent/CorfuQueueTxTest.java

```diff
@@ -88,11 +93,13 @@ public Record(CorfuRecordId id, String data) {
                     TXBegin(txnType);
                     Long coinToss = new Random().nextLong() % numConflictKeys;
                     conflictMap.put(coinToss, coinToss);
+                    corfuQueue.enqueue(queueData);
+                    // Each transaction may or may not sleep to simulate out of order between enQ & commit
+                    TimeUnit.MILLISECONDS.sleep(coinToss);
                     lock.lock();
```

Comment:
We should remove the lock and generated some tx abort scenarios. You can define validator as a hashtable instead of an array. I am not clear if tx aborted, if the record will be removed from the queue or not?

Reply:
Aborts are definitely happening because we conflict on one of two values, for example in a test with 2000 transactions only about 1500 succeed. But I will add an assert to verify this too.

Reply:
Actually it turns out travis sometimes serializes the runs which means there might not be any aborts due to concurrency. So I can't add an assert or test will become very non-deterministic :(